runtime/common/src/impls.rs

```diff
@@ -163,3 +166,139 @@ impl From<u128> for InstanceId {
 		Self(v)
 	}
 }
+
+#[derive(codec::Encode, codec::Decode, TypeInfo, Debug, Clone, Eq, PartialEq)]
+pub(crate) struct TrancheInvestorInfo {
+	pub(crate) tranche_id: TrancheId,
+	pub(crate) permissioned_at: Moment,
+}
+
+impl<MaxTranches: Get<TrancheId>, MaxHold, MinDelay> Default
+	for TrancheInvestors<MaxHold, MinDelay, MaxTranches>
```

Comment:
We dont need to MaxHold. Just Mindelay so that Investor has time to do some redeem if required. For some pools, we may have the max value of the moment

Reply:
I thought we defined it as

So the check `has_permission(pool_id, who, PoolRole::TrancheInvestor(tranche_id)` must check if the whitelisting could be expired.
And the check `rm_permission(pool_id, who, PoolRole::TrancheInvestor(tranche_id)` must check if the minimum delay has been passed.

`MaxHold`represents the first one, `MinDelay` the last one

Reply:
HasPermission should ensure the this instant a investor can invest and hold tokens. Which can be simple check of `permissioned_at > now`

RmPermission needs to ensure the new permissioned_at moment is atleast more than MinDelay so that investor has enough time to redeem or make other investing decisions